modules/varscan/1.0/varscan.smk

```diff
@@ -0,0 +1,194 @@
+    log:
+        stderr = CFG["logs"]["mpileup"] + "{seq_type}--{genome_build}/{sample_id}.bam2mpu.stderr.log"
+    params:
+        opts = CFG["options"]["mpileup"]
+    conda:
+        CFG["conda_envs"]["samtools"]
+    threads:
+        CFG["threads"]["mpileup"]
+    resources:
+        mem_mb = CFG["mem_mb"]["mpileup"]
+    shell:
+        op.as_one_line("""
+        samtools mpileup {params.opts}
+        -f {input.fasta} {input.bam}
+        > {output.mpu}
+        2> {log.stderr}
+        """)
+
+
+rule _varscan_somatic:
+    input:
+        tumour_mpu = CFG["dirs"]["mpileup"] + "{seq_type}--{genome_build}/{tumour_id}.mpileup",
+        normal_mpu = CFG["dirs"]["mpileup"] + "{seq_type}--{genome_build}/{normal_id}.mpileup"
+    output:
+        snp = CFG["dirs"]["varscan"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/snp.vcf",
+        indel = CFG["dirs"]["varscan"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/indel.vcf"
+    wildcard_constraints:
+        pair_status = "matched|unmatched"
+    log:
+        stdout = CFG["logs"]["varscan"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/varscan_somatic.stdout.log",
+        stderr = CFG["logs"]["varscan"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/varscan_somatic.stderr.log"
+    params:
+        opts = op.switch_on_wildcard("seq_type", CFG["options"]["somatic"])
+    conda:
+        CFG["conda_envs"]["varscan"]
+    threads:
+        CFG["threads"]["somatic"]
+    resources:
+        mem_mb = CFG["mem_mb"]["somatic"]
+    shell:
+        op.as_one_line("""
+        varscan somatic 
+        {input.normal_mpu} {input.tumour_mpu} 
+        --output-snp {output.snp} --output-indel {output.indel}
+        {params.opts}
+        > {log.stdout} 2> {log.stderr}
+        """)
+
+
+rule _varscan_unpaired:
+    input:
+        tumour_mpu = CFG["dirs"]["mpileup"] + "{seq_type}--{genome_build}/{tumour_id}.mpileup",
+    output:
+        vcf = CFG["dirs"]["varscan"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{vcf_name}.vcf"
+    wildcard_constraints:
+        pair_status = "no_normal"
+    log:
+        stderr = CFG["logs"]["varscan"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/varscan_{vcf_name}.stderr.log"
+    params:
+        opts = op.switch_on_wildcard("seq_type", CFG["options"]["unpaired"]),
+        cns = op.switch_on_wildcard("seq_type", {"cns": "--variants", "indel": "", "snp": ""})
+    conda:
+        CFG["conda_envs"]["varscan"]
+    threads:
+        CFG["threads"]["unpaired"]
+    resources:
+        mem_mb = CFG["mem_mb"]["unpaired"]
+    shell:
+        op.as_one_line("""
+        varscan mpileup2{wildcards.vcf_name} 
+        {input.tumour_mpu} 
+        {params.opts}
+        {params.cns}
+        > {output.vcf} 2> {log.stderr}
+        """)
+
+
+rule _varscan_symlink_maf:
```

Comment:
Should this be called _varscan_symlink_vcf?

Reply:
Actually, what does this do? Seems redundant with the rule below

Reply:
This part is a link to the `vcf2maf` module. I wrote the `vcf2maf` module similar to the `utils` module where it's just a collection of rules. It allows it to be used by multiple modules without making the code too complicated. 

So `_varscan_symlink_maf` symlinks the vcf file into a directory (`CFG["dirs"]["maf"]`), where vcf2maf is  executed.